tests/integration/cattletest/core/test_svc_discovery.py

```diff
@@ -1457,10 +1457,10 @@ def test_global_service_update_label(new_context):
     instance2_host = instance2.hosts()[0].id
     assert instance2_host == host2.id
 
-    # destroy the instance, reactivate the service and check
-    # both hosts got instances
-    _instance_remove(instance1, client)
+    # deactivate the service, destroy the instance,
+    # reactivate the service, and check both hosts got instances
     service = wait_state(client, service.deactivate(), 'inactive')
+    _instance_remove(instance1, client)
```

Comment:
@cjellick this test seems is failing on drone. service.deactivate just schedules instances removal, but doesn't wait for stop to finish. So calling instance.remove on instance that is not stopped, results in error.